runtime/compiler/p/runtime/J9PPCArrayTranslate.spp

```diff
@@ -1313,973 +1264,9 @@ __arrayTranslateTROT:
-	rlwinm	r0,r12,29,29,30		! r0 = (((len - headerCount) / 16) & 3) << 1	blockCount << 1
-	srwi	r11,r12,6		! r11 = (len - headerCount) / 16		block4Count
-	neg	r7,r10			! r7 = -headerCount
-	slwi	r12,r3,6		! r12 = (swSlowPath << 6)
-	srwi	r3,r4,1			! inp >> 1
-	xor	r3,r3,r5		! (inp >> 1) & outp
-	andi.	r3,r3,7			! ((inp >> 1) XOR outp) & 7
-	mfcr	r3			! CR0_EQ = (((inp >> 1) XOR outp) & 7) == 0
-	rlwinm	r3,r3,8,26,26
-	or	r12,r12,r3		! r12 = (swSlowPath << 6) | (swSrcDstAlign << 5)
-	rlwinm	r3,r7,5,27,27		! r3 = (headerCount != 0) << 4			swHeader << 4
-	or	r9,r12,r0		! r9 = (swSlowPath << 6) | (blockCount << 1)
-	rlwinm	r12,r8,0,28,31		! r12 = (outp + len) & 15			trailerCount
-	neg	r8,r11			! r8 = -blockCount
-	or	r3,r9,r3		! r3 = (swSlowPath << 6) | (swHeader << 4) | (blockCount << 1)
-	rlwinm	r0,r8,4,28,28		! r0 = (blockCount != 0) << 3			swBlock4Loop << 3
-	neg	r7,r12			! r7 = -trailerCount
-	or	r8,r3,r0	    	! r8 = (swSlowPath << 6) | (swHeader << 4) | (swBlock4Loop << 3) | (blockCount << 1)
-	rlwinm	r9,r7,1,31,31		! r0 = (trailerCount != 0) << 0			swTrailer << 0
-	or	r0,r8,r9	    	! r0 = (swSlowPath << 6) | (swHeader << 4) | (swBlock4Loop << 3) | (blockCount << 1) | (swTrailer << 0)
-	mfctr	r7			! Restore maxChar
-
-	! Look up a function pointer from dispatch table CTR <- func[r0]
-	laddr	r3, J9TR_VMThreadRTOCOffset(J9VM_STRUCT)	! Restore TOC/GOT
-#ifdef TR_HOST_64BIT
-	rlwinm	r0, r0, 3, 0xFFFFFFF8
-#else
-	rlwinm	r0, r0, 2, 0xFFFFFFFC
-#endif
-#ifdef AIXPPC
-	laddr	r3, TOC__arrayTranslateLabelTable(r3)
-#elif defined(LINUXPPC64)
-	laddr	r3, TOC__arrayTranslateLabelTable@toc(r3)
-#else
-	laddr	r3, __arrayTranslateLabelTable@got(r3)
-#endif
-	laddrx	r3, r3, r0
-	mtctr	r3
-	! Do rest of initialization
-    	li 	r3, 0			! index = 0
-	li 	r8, 16			! const16 = 16
-	li 	r9, 32			! const32 = 32
-	li 	r0, 48			! const48 = 48
-	!-------------------------------
-    	! Exit condition
-	!   reg		content
-	!-------------------------------
-	!   r0  -- 	const48
-	!   r3  -- 	index
-	!   r4  -- 	inp
-	!   r5  -- 	outp
-	!   r6  -- 	len
-	!   r7  -- 	maxChar
-	!   r8  -- 	const16
-	!   r9  -- 	const32
-	!   r10 -- 	headerCount
-	!   r11 --	blockCount
-	!   r12 --	trailerCount
-	!   vr0 --	vmaxChars
-	!-------------------------------
-	bctr
-!----------------------------------------------------------------------
-! end of arrayTranslateTRTOSimpleVMX
-!----------------------------------------------------------------------
-
-!----------------------------------------------------------------------
-! macro definitions
-!    macroArrayTranslate swHeader,swSrcDstAlign,swBlock4Loop,swBlock2,swBlock1,swTrailer
-!      macroArrayTranslateHeader
-!      macroArrayTranslateUnrollInit
-!      macroArrayTranslateUnrolledBodyN swSrcDstAlign num
-!        macroArrayTranslateUnrolledBodyGen vin0 vin2 swSrcDstAlign outReg firstBlock
-!      macroArrayTranslateTrailer
-!----------------------------------------------------------------------
```

Comment:
From here downward it looks like you are also getting rid of a bunch of macro definitions that appear to be unrelated to `arrayTranslateTRTOSimpleVMX`. They are behind an `#if 0` so I'm not sure if they are really needed anymore. But you should check to make sure.

Reply:
@IBMJimmyk thanks for the feedback. My thought was, because they were not used for 2 yo+, maybe they are good candidate for cleanup. Also, because they were defined within the .startproc and .endproc.